erdgo/proxy.go

```diff
@@ -0,0 +1,223 @@
+func NewElrondProxy(url string) *ElrondProxy {
+	ep := &ElrondProxy{
+		proxyURL: url,
+	}
+
+	return ep
+}
+
+// GetNetworkConfig retrieves the network configuration from the proxy
+func (ep *ElrondProxy) GetNetworkConfig() (*NetworkConfig, error) {
+	data, err := ep.getHTTP(networkConfigEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	response := &NetworkConfigResponse{}
+	err = json.Unmarshal(data, response)
+	if err != nil {
+		return nil, err
+	}
+	if response.Error != "" {
+		return nil, errors.New(response.Error)
+	}
+
+	return response.Data.Config, nil
+}
+
+// GetAccount retrieves an account info from the network (nonce, balance)
+func (ep *ElrondProxy) GetAccount(address string) (*Account, error) {
+	if !IsValidBech32Address(address) {
+		return nil, errInvalidAddress
+	}
+	endpoint := fmt.Sprintf(accountEndpoint, address)
+	data, err := ep.getHTTP(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	response := &AccountResponse{}
+	err = json.Unmarshal(data, response)
+	if err != nil {
+		return nil, err
+	}
+	if response.Error != "" {
+		return nil, errors.New(response.Error)
+	}
+
+	return response.Data.Account, nil
+}
+
+// SendTransaction broadcasts a transaction to the network and returns the txhash if successful
+func (ep *ElrondProxy) SendTransaction(tx *Transaction) (string, error) {
+	jsonTx, err := json.Marshal(tx)
+	if err != nil {
+		return "", err
+	}
+	data, err := ep.postHTTP(sendTransactionEndpoint, jsonTx)
+	if err != nil {
+		return "", err
+	}
+	response := &SendTransactionResponse{}
+	err = json.Unmarshal(data, response)
+	if err != nil {
+		return "", err
+	}
+	if response.Error != "" {
+		return "", errors.New(response.Error)
+	}
+
+	return response.Data.TxHash, nil
+}
+
+// GetTransactionStatus retrieves a transaction's status from the network
+func (ep *ElrondProxy) GetTransactionStatus(hash string) (string, error) {
+	endpoint := fmt.Sprintf(getTransactionStatusEndpoint, hash)
+	data, err := ep.getHTTP(endpoint)
+	if err != nil {
+		return "", err
+	}
+	response := &TransactionStatus{}
+	err = json.Unmarshal(data, response)
+	if err != nil {
+		return "", err
+	}
+	if response.Error != "" {
+		return "", errors.New(response.Error)
+	}
+
+	return response.Data.Status, nil
+}
+
+// GetTransactionInfo retrieves a transaction's details from the network
+func (ep *ElrondProxy) GetTransactionInfo(hash string) (*TransactionInfo, error) {
+	endpoint := fmt.Sprintf(getTransactionInfoEndpoint, hash)
+	data, err := ep.getHTTP(endpoint)
+	if err != nil {
+		return nil, err
+	}
+	response := &TransactionInfo{}
+	err = json.Unmarshal(data, response)
+	if err != nil {
+		return nil, err
+	}
+	if response.Error != "" {
+		return nil, errors.New(response.Error)
+	}
+
+	return response, nil
+}
+
+// GetLatestHyperblockNonce retrieves the latest hyperblock (metachain) nonce from the network
+func (ep *ElrondProxy) GetLatestHyperblockNonce() (uint64, error) {
+	endpoint := fmt.Sprintf(getNetworkStatusEndpoint, metachainID)
```

Comment:
this should be latest Hyperblock nonce endpoint

Reply:
The hyperblock nonce is actually the metachain nonce.